File: clean.py
import os

def delete_venv(root_dir):
    for dirpath, dirnames, filenames in os.walk(root_dir):
        for dirname in dirnames[:]:
            if dirname == 'venv':
                venv_path = os.path.join(dirpath, dirname)
                print(f"Deleting: {venv_path}")
                delete_folder_recursive(venv_path)
                dirnames.remove(dirname)

def delete_folder_recursive(folder_path):
    if os.path.exists(folder_path):
        for root, dirs, files in os.walk(folder_path, topdown=False):
            for file in files:
                file_path = os.path.join(root, file)
                os.remove(file_path)
            for dir in dirs:
                dir_path = os.path.join(root, dir)
                os.rmdir(dir_path)
        os.rmdir(folder_path)
        print(f"Deleted: {folder_path}")

# Get the root directory dynamically
root_directory = os.path.dirname(os.path.abspath(__file__))

delete_venv(root_directory)

File: clean.java
import java.io.File;

public class clean {

    public static void deleteTargetFolders(File rootDir) {
        File[] files = rootDir.listFiles();

        if (files != null) {
            for (File file : files) {
                if (file.isDirectory()) {
                    if (file.getName().equals("target")) {
                        System.out.println("Deleting: " + file.getAbsolutePath());
                        deleteFolderRecursive(file);
                    } else {
                        deleteTargetFolders(file);
                    }
                }
            }
        }
    }

    public static void deleteFolderRecursive(File folder) {
        File[] files = folder.listFiles();

        if (files != null) {
            for (File file : files) {
                if (file.isDirectory()) {
                    deleteFolderRecursive(file);
                } else {
                    file.delete();
                }
            }
        }

        folder.delete();
    }

    public static void main(String[] args) {
        // Get the root directory dynamically
        String rootDirectory = System.getProperty("user.dir");

        File rootDir = new File(rootDirectory);
        deleteTargetFolders(rootDir);
    }
}
